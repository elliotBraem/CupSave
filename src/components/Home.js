import React from 'react';

import {View, Text, StyleSheet} from 'react-native';

const Home = ({todos, addTodo, onNewTodoChange, newTodoText}) => (
  <View style={styles.container}>
    <Text style={styles.header}>Hello</Text>
  </View>
);

export default Home;

const styles = StyleSheet.create({
  container: {
    marginTop: 100,
    paddingLeft: 15,
    paddingRight: 15,
  },
  header: {
    alignSelf: 'center',
    marginBottom: 50,
  },
});
