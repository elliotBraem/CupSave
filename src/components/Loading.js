import React from 'react';
import {View, ActivityIndicator} from 'react-native';
import COLORS from '../constants/colors';

const Loading = () => (
  <View style={{flex: 1, flexDirection: 'column', justifyContent: 'center'}}>
    <ActivityIndicator size="large" color={COLORS.primary} />
  </View>
);

export default Loading;
