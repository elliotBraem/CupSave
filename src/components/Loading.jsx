import React from 'react';
import {View, ActivityIndicator} from 'react-native';

const Loading = () => (
  <View style={{flex: 1, flexDirection: 'column', justifyContent: 'center'}}>
    <ActivityIndicator size="large" color="blue" />
  </View>
);

export default Loading;
