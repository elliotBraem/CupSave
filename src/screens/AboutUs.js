import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
import {StyleSheet, View, Platform} from 'react-native';
import CustomHeader from '../components/CustomHeader';
import COLORS from '../constants/colors';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'space-between',
    backgroundColor: COLORS.primary,
  },
  inner: {
    marginTop: Platform.OS === 'ios' ? 100 : 100 - 24,
  },
});

class AboutUsScreen extends PureComponent {
  static propTypes = {
    navigation: PropTypes.shape({
      openDrawer: PropTypes.func.isRequired,
      navigate: PropTypes.func.isRequired,
    }).isRequired,
  };

  render() {
    return (
      <View style={styles.container}>
        <CustomHeader title="About Us" />
        <View style={styles.header} />
      </View>
    );
  }
}

export default AboutUsScreen;
