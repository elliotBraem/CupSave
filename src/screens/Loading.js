import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {View, Text, ActivityIndicator, StyleSheet} from 'react-native';
import {withFirebase} from 'react-redux-firebase';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
});

class LoadingScreen extends Component {
  constructor(props) {
    super(props)
  }

  static propTypes = {
    navigation: PropTypes.shape({
      navigate: PropTypes.func.isRequired,
    }).isRequired,
    firebase: PropTypes.shape({
      auth: PropTypes.func.isRequired,
    }).isRequired, // from withFirebase
  };

  componentDidMount() {
    const {firebase, navigation} = this.props;
    firebase.auth().onAuthStateChanged(user => {
      navigation.navigate(user ? 'App' : 'Auth');
    });
  }

  render() {
    return (
      <View style={styles.container}>
        <Text>Loading</Text>
        <ActivityIndicator size="large" />
      </View>
    );
  }
}

export default withFirebase(LoadingScreen);
