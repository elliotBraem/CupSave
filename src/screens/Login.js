import React, {Component} from 'react';
import {Text, StyleSheet, View, Alert} from 'react-native';
import PropTypes from 'prop-types';
import {withFirebase} from 'react-redux-firebase';
import {Button, Input, H1, H4, P} from 'nachos-ui';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
  },
  // buttons: {
  //   alignItems: 'center',
  //   justifyContent: 'center',
  //   marginBottom: 10,
  // },
  header: {
    fontSize: 38,
    // marginBottom: 20,
  },
  // textInput: {
  //   height: 40,
  //   width: '100%',
  //   borderColor: 'gray',
  //   borderWidth: 1,
  //   marginTop: 14,
  //   alignSelf: 'stretch',
  //   textAlign: 'center',
  // },
  inputStyle: {
    margin: 15,
  },
  btnStyle: {
    margin: 15,
  },
});

class LoginScreen extends Component {
  static navigationOptions = {
    title: 'Login',
  };

  static propTypes = {
    navigation: PropTypes.shape({
      openDrawer: PropTypes.func.isRequired,
      navigate: PropTypes.func.isRequired,
    }).isRequired,
    firebase: PropTypes.shape({
      login: PropTypes.func.isRequired,
    }).isRequired, // from withFirebase
    auth: PropTypes.object, // from withFirebase
  };

  state = {email: '', password: '', errorMessage: null};

  handleLogin = () => {
    const {email, password} = this.state;
    const {navigation, firebase} = this.props;

    if (email.trim() === '' || password.trim() === '') {
      Alert.alert('Invalid Parameters:', 'Username / password cannot be empty');
    } else {
      firebase
        .login({
          email,
          password,
        })
        .then(() => navigation.navigate('Home'))
        .catch(error => this.setState({errorMessage: error.message}));
      console.log('handleLogin');
    }
  };

  render() {
    const {navigation, firebase} = this.props;

    return (
      <View style={styles.container}>
        <H1 style={styles.header} align="center">
          Welcome to{'\n'}CupSave!
        </H1>
        <H4>Let&#39;s get started</H4>
        <H4>Login</H4>
        {this.state.errorMessage && <Text style={{color: 'red'}}>{this.state.errorMessage}</Text>}
        <Input
          style={styles.inputStyle}
          autoCapitalize="none"
          placeholder="Email"
          value={this.state.email}
          onChangeText={email => this.setState({email})}
        />
        <Input
          secureTextEntry
          style={styles.inputStyle}
          autoCapitalize="none"
          placeholder="Password"
          onChangeText={password => this.setState({password})}
          value={this.state.password}
        />
        <Button style={styles.btnStyle} onPress={this.handleLogin}>
          Login
        </Button>
        <P>Don&#39;t have an account?</P>
        <Button onPress={() => navigation.navigate('SignUp')} style={styles.button}>
          Sign Up
        </Button>
        <Button onPress={() => navigation.openDrawer()} style={styles.button}>
          Open drawer
        </Button>
        {/* <Button title="Open drawer" onPress={() => navigation.openDrawer()} style={styles.button} />
        <Button title="Go back" onPress={() => navigation.navigate('Home')} style={styles.button} /> */}
      </View>
    );
  }
}

export default withFirebase(LoginScreen);
