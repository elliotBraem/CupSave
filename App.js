/* global window */
import React from 'react';
import {Provider} from 'react-redux';
import {PersistGate} from 'redux-persist/integration/react';
import {AppRegistry} from 'react-native';

import {ReactReduxFirebaseProvider} from 'react-redux-firebase';
import {createFirestoreInstance} from 'redux-firestore';

import firebase from 'firebase/app';
import 'firebase/auth';
import 'firebase/firestore';
import 'firebase/storage';

import {firebaseConfig, reactReduxFirebaseConfig} from './src/constants/firebase';
import configureStore from './src/store/configureStore';
import Loading from './src/components/Loading';
import AppContainer from './src/navigation/AppNavigation';

const initialState = window.__INITIAL_STATE__ || {
  firebase: {
    authError: null,
  },
};

const {store, persistor} = configureStore(initialState);

// Initialize Firebase
// Initialize Firebase only if an fbInstance was not passed to the window (tests)
if (!window.fbInstance) {
  firebase.initializeApp(firebaseConfig);
}

// Initialize firestore service on firebase instance
firebase.firestore();

const App = () => (
  <Provider store={store}>
    <ReactReduxFirebaseProvider
      config={reactReduxFirebaseConfig}
      dispatch={store.dispatch}
      firebase={firebase}
      createFirestoreInstance={createFirestoreInstance}>
      <PersistGate loading={<Loading />} persistor={persistor}>
        <AppContainer />
      </PersistGate>
    </ReactReduxFirebaseProvider>
  </Provider>
);

AppRegistry.registerComponent('CupSave', () => App, false);

export default App;
