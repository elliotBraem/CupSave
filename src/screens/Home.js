import React, {PureComponent} from 'react';
import {StyleSheet, View, Platform} from 'react-native';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import SaveCupForm from '../components/SaveCupForm';
import CustomHeader from '../components/CustomHeader';
import COLORS from '../constants/colors';
import * as authActions from '../store/actions/auth';
import Loading from '../components/Loading';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'flex-start',
    flexDirection: 'column',
    alignItems: 'center',
    backgroundColor: COLORS.primary,
  },
  inner: {
    marginTop: Platform.OS === 'ios' ? 100 : 100 - 24,
  },
});

class HomeScreen extends PureComponent {
  static propTypes = {
    incrementConsumption: PropTypes.func.isRequired,
    auth: PropTypes.shape({
      isAuthenticated: PropTypes.bool.isRequired,
    }).isRequired,
    fetchAuthData: PropTypes.func.isRequired,
  };

  componentDidMount = () => {
    const {fetchAuthData, auth} = this.props;

    if (!auth.isAuthenticated) {
      fetchAuthData();
    }
  };

  render() {
    const {auth, incrementConsumption} = this.props;

    if (!auth.isLoaded) {
      return <Loading />;
    }

    return (
      <View style={styles.container}>
        <CustomHeader title="Home" style={styles.header} />
        <View style={styles.inner}>
          <SaveCupForm onSaveCupFormSubmit={incrementConsumption} />
        </View>
      </View>
    );
  }
}

const mapDispatchToProps = (dispatch, ownProps) => {
  return {
    incrementConsumption: () => dispatch(authActions.dbIncrementConsumption()),
    fetchAuthData: () => dispatch(authActions.dbOnAuthorizeStateChange()),
  };
};

const mapStateToProps = (state, ownProps) => {
  const auth = state.auth || {};

  return {
    auth,
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(HomeScreen);
