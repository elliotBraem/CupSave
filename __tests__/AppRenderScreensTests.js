import React from 'react';
import renderer from 'react-test-renderer';

import App from '../App';
import AboutUs from '../src/screens/AboutUs';
import {HomeScreen} from '../src/screens/Home';
import {LoadingScreen} from '../src/screens/Loading';
import {LoginScreen} from '../src/screens/Login';
import {MapScreen} from '../src/screens/Map';
import {ProfileScreen} from '../src/screens/Profile';
import {QRScannerScreen} from '../src/screens/QRScanner';
import Settings from '../src/screens/Settings';
import {SignUpScreen} from '../src/screens/SignUp';

// App
it('App renders without crashing', () => {
  const rendered = renderer.create(<App />).toJSON();
  expect(rendered).toBeTruthy();
});

it('App test against snapshot', () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});

// Screens -> AboutUs
it('AboutUs renders without crashing', () => {
  const rendered = renderer.create(<AboutUs />).toJSON();
  expect(rendered).toBeTruthy();
});

it('AboutUs test against snapshot', () => {
  const tree = renderer.create(<AboutUs />).toJSON();
  expect(tree).toMatchSnapshot();
});

// Screens -> Home
it('Home renders without crashing', () => {
  console.error = jest.fn();
  const rendered = renderer.create(<HomeScreen auth={{user: {}, isLoaded: true}} />).toJSON();
  expect(rendered).toBeTruthy();
  expect(console.error).toHaveBeenCalled();
});

it('Home test against snapshot', () => {
  const tree = renderer.create(<HomeScreen auth={{user: {}, isLoaded: true}} />).toJSON();
  expect(tree).toMatchSnapshot();
});

// Screens -> Loading
it('Loading renders without crashing', () => {
  const rendered = renderer
    .create(<LoadingScreen auth={{isAuthenticated: true}} navigation={{navigate: () => {}}} />)
    .toJSON();
  expect(rendered).toBeTruthy();
});

it('Loading test against snapshot', () => {
  const tree = renderer
    .create(<LoadingScreen auth={{isAuthenticated: true}} navigation={{navigate: () => {}}} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

// Screens -> Login
it('Login renders without crashing', () => {
  console.error = jest.fn();
  const rendered = renderer.create(<LoginScreen />).toJSON();
  expect(rendered).toBeTruthy();
  expect(console.error).toHaveBeenCalled();
});

it('Login test against snapshot', () => {
  const tree = renderer.create(<LoginScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});

// Screens -> Map
it('Map renders without crashing', () => {
  console.error = jest.fn();
  const rendered = renderer.create(<MapScreen locations={{isLoaded: true}} />).toJSON();
  expect(rendered).toBeTruthy();
  expect(console.error).toHaveBeenCalled();
});

it('Map test against snapshot', () => {
  const tree = renderer.create(<MapScreen locations={{isLoaded: true}} />).toJSON();
  expect(tree).toMatchSnapshot();
});

// Screens -> Profile
it('Profile renders without crashing', () => {
  console.error = jest.fn();
  const rendered = renderer.create(<ProfileScreen auth={{user: {}, isLoaded: true}} />).toJSON();
  expect(rendered).toBeTruthy();
  expect(console.error).toHaveBeenCalled();
});

it('Profile test against snapshot', () => {
  const tree = renderer.create(<ProfileScreen auth={{user: {}, isLoaded: true}} />).toJSON();
  expect(tree).toMatchSnapshot();
});

// Screens -> QRScanner
it('QRScanner renders without crashing', () => {
  console.error = jest.fn();
  const rendered = renderer.create(<QRScannerScreen />).toJSON();
  expect(rendered).toBeTruthy();
  expect(console.error).toHaveBeenCalled();
});

it('QRScanner test against snapshot', () => {
  const tree = renderer.create(<QRScannerScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});

// Screens -> Settings
it('Settings renders without crashing', () => {
  const rendered = renderer.create(<Settings />).toJSON();
  expect(rendered).toBeTruthy();
});

it('Settings test against snapshot', () => {
  const tree = renderer.create(<Settings />).toJSON();
  expect(tree).toMatchSnapshot();
});

// Screens -> SignUp
it('SignUp renders without crashing', () => {
  console.error = jest.fn();
  const rendered = renderer.create(<SignUpScreen />).toJSON();
  expect(rendered).toBeTruthy();
  expect(console.error).toHaveBeenCalled();
});

it('SignUp test against snapshot', () => {
  const tree = renderer.create(<SignUpScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});
