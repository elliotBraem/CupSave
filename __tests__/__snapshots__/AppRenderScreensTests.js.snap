// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App test against snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#79db85"
    hidesWhenStopped={true}
    size="large"
  />
</View>
`;

exports[`Loading test against snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text>
    Loading
  </Text>
  <ActivityIndicator
    animating={true}
    color="#999999"
    hidesWhenStopped={true}
    size="large"
  />
</View>
`;

exports[`Login test against snapshot 1`] = `
<View
  enable={true}
  onLayout={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffffff",
        "borderTopColor": "#ffffff",
        "borderTopWidth": 50,
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "flex-start",
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#1A1A1A",
        "borderRadius": 40,
        "borderWidth": 2,
        "flex": 0.6,
        "flexDirection": "row",
        "justifyContent": "center",
        "marginHorizontal": "10%",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1.5,
          "marginLeft": "5%",
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "flex": 1,
            "height": null,
            "resizeMode": "contain",
            "width": null,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "space-around",
          "margin": "5%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#79db85",
            "fontSize": 22,
            "textAlign": "center",
          }
        }
      >
         Save a cup 
      </Text>
      <Text
        style={
          Object {
            "color": "#ffffff",
            "fontSize": 22,
            "fontStyle": "italic",
            "textAlign": "center",
          }
        }
      >
         Save the world 
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 0.1,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
        }
      }
    >
      Let's get started
    </Text>
  </View>
  <View
    style={
      Object {
        "alignSelf": "center",
        "flex": 0.8,
        "justifyContent": "space-between",
        "width": "80%",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      onChangeText={[Function]}
      placeholder="Email"
      style={
        Object {
          "alignSelf": "stretch",
          "backgroundColor": "#ffffff",
          "borderColor": "gray",
          "borderWidth": 1,
          "height": 40,
          "marginBottom": "2%",
          "textAlign": "center",
          "width": "100%",
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      onChangeText={[Function]}
      placeholder="Password"
      secureTextEntry={true}
      style={
        Object {
          "alignSelf": "stretch",
          "backgroundColor": "#ffffff",
          "borderColor": "gray",
          "borderWidth": 1,
          "height": 40,
          "marginBottom": "2%",
          "textAlign": "center",
          "width": "100%",
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      style={
        Object {
          "flex": 0.5,
          "height": 40,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityStates={Array []}
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
                Object {
                  "color": "#79db85",
                },
              ]
            }
          >
            Login
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "height": 40,
        }
      }
    />
    <View
      style={
        Object {
          "flex": 0.5,
          "height": 40,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityStates={Array []}
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
                Object {
                  "color": "#3b5998",
                },
              ]
            }
          >
            Login with Facebook
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "marginTop": "2%",
        }
      }
    >
      <Text
        style={
          Object {
            "alignSelf": "center",
            "margin": 15,
          }
        }
      >
        Don't have an account?
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 0.5,
          "height": 40,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityStates={Array []}
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
                Object {
                  "color": "#79db85",
                },
              ]
            }
          >
            Sign Up
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`QRScanner test against snapshot 1`] = `
<Text>
  Requesting for camera permission
</Text>
`;

exports[`SignUp test against snapshot 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffffff",
        "borderTopColor": "#ffffff",
        "borderTopWidth": 50,
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "flex-start",
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#1A1A1A",
        "borderRadius": 40,
        "borderWidth": 2,
        "flex": 0.5,
        "flexDirection": "row",
        "justifyContent": "center",
        "marginHorizontal": "10%",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1.5,
          "marginLeft": "5%",
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "flex": 1,
            "height": null,
            "resizeMode": "contain",
            "width": null,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "space-around",
          "margin": "5%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#79db85",
            "fontSize": 22,
            "textAlign": "center",
          }
        }
      >
         Welcome to CupSave! 
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 0.05,
        "justifyContent": "center",
      }
    }
  />
  <View
    style={
      Object {
        "alignSelf": "center",
        "flex": 0.8,
        "justifyContent": "space-between",
        "width": "80%",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      onChangeText={[Function]}
      placeholder="Email"
      style={
        Object {
          "alignSelf": "stretch",
          "backgroundColor": "#ffffff",
          "borderColor": "gray",
          "borderWidth": 1,
          "height": 40,
          "marginBottom": "2%",
          "textAlign": "center",
          "width": "100%",
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      onChangeText={[Function]}
      placeholder="Password"
      secureTextEntry={true}
      style={
        Object {
          "alignSelf": "stretch",
          "backgroundColor": "#ffffff",
          "borderColor": "gray",
          "borderWidth": 1,
          "height": 40,
          "marginBottom": "2%",
          "textAlign": "center",
          "width": "100%",
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      onChangeText={[Function]}
      placeholder="Confirm Password"
      secureTextEntry={true}
      style={
        Object {
          "alignSelf": "stretch",
          "backgroundColor": "#ffffff",
          "borderColor": "gray",
          "borderWidth": 1,
          "height": 40,
          "marginBottom": "2%",
          "textAlign": "center",
          "width": "100%",
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      style={
        Object {
          "flex": 0.5,
          "height": 40,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityStates={Array []}
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
                Object {
                  "color": "#79db85",
                },
              ]
            }
          >
            Sign Up
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "height": 40,
        }
      }
    />
    <View
      style={
        Object {
          "flex": 0.5,
          "height": 40,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityStates={Array []}
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
                Object {
                  "color": "#3b5998",
                },
              ]
            }
          >
            Sign up with Facebook
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 0.3,
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "alignSelf": "center",
          }
        }
      >
        Already have an account?
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 0.5,
          "height": 40,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityStates={Array []}
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
                Object {
                  "color": "#79db85",
                },
              ]
            }
          >
            Back to login
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
